server/src/app.rs

```diff
@@ -80,14 +93,14 @@ impl Application {
         let (shutdown_signal, shutdown_watch) = drain::channel();
         // start node ctrl service base
         let mut node_ctrl_handle = tokio::spawn(self.node_ctrl.run(shutdown_watch.clone()));
-
         // Init the meta. This will reload the schemas in memory.
         self.meta.init().await?;
-
         let worker_command_tx = self.worker.worker_command_tx();
         let mut meta_handle =
             tokio::spawn(self.meta.run(shutdown_watch.clone(), worker_command_tx));
-        let mut worker_handle = tokio::spawn(self.worker.run(shutdown_watch));
+        // Ideally we should not pass the AdminServer to worker since it's a separate role,
+        // but we have to do that until we allow worker storage state to be queried externally/remotely.
+        let mut worker_handle = tokio::spawn(self.worker.run(shutdown_watch, self.admin));
```

Comment:
An alternative could be to let the `worker` give us the `storage_query_context` here (as very simplistic form of exposing worker state to the outside). That way, the `worker` wouldn't have to know about the `admin` crate.

Reply:
I found it a bit hairy to do that, especially that we only create the storage context when we run the worker which means that we need async mechanism to pass the context back. 

In all cases, we know that this is not the design we want and there is no way at the moment to run admin server on a  node that's not a worker. We will change this in the near future.

Reply:
Aren't we creating the storage context when creating the worker?

Reply:
@tillrohrmann ignore me, I confused your suggestion with another issue I faced. Good call, sending an update now.